### DISCUSSION TITLE
Run Siri Shortcuts actions in the background

### DISCUSSION THREAD

**Discussion by fluxor99232** [BLeeEZ/amperfy#253] [Q&A]
Is it possible to run the Amperfy actions in the background without opening Amperfy (like actions for the Music app)? Currently I can't run any Shortcuts containing Amperfy actions without unlocking my iPhone, meaning I can't trigger them from my HomePod like I usually do.

**Comment by BLeeEZ** [comment]
I didn't found a possibility to do that. If you or someone else can show me how do that I am happy to implement that.

**Reply by fluxor99232** [reply]
That would be amazing! For example the popular radio streaming app "Receiver" ([https://www.mamp.info/receiver/en/](url)) has that capability (it has a "Show When Run" toggle for all action). :-)

**Comment by BLeeEZ** [comment]
All shortcuts are now processed in background: https://github.com/BLeeEZ/amperfy/commit/8d9486024b6899a22aa7482d4a41e113c73aa41e

**Reply by fluxor99232** [reply]
Awesome! Can’t wait to give it a try!

**Comment by BLeeEZ** [comment]
This feature can be tested in the newest beta v1.2.1 (2): https://github.com/BLeeEZ/amperfy/issues/25

**Reply by fluxor99232** [reply]
Works great! Have been able to trigger a shortcut from my HomePod without my phone being unlocked! 👌
